### DISCUSSION TITLE
Release cadence

### DISCUSSION THREAD

**Discussion by adriangb** [psycopg/psycopg#812] [General]
Is there any information on psycopg's release cadence? The last release was on Febuary 4th, over 3 months ago.

**Comment by dvarrazzo** [comment]
It happens that I've just finished to fix bitrotten CI and to close a couple of complex tickets, so the next release is imminent.

In general, no, there is no guaranteed release cadence for psycopg. It depends very much on the bugs being fixed and by my personal workload. Since 3.1.18 release we have made [a considerable amount of work to prepare 3.2](https://github.com/psycopg/psycopg/compare/e00a7a5e..4735eb02), while in 3.1 there weren't critical bugs open, and still [there was a lot of work done there too](https://github.com/psycopg/psycopg/compare/8585a23f..040489c6) to get to 3.1.19.

At the moment I have paying customers which require attention. Free software pays very little, therefore it can only take a proportional amount of time. This sentence is absolutely false, by the way: it takes my attention and my time much more than what it pays financially. Then you meet people like the dude in #807 which makes me think that giving away any work for free is not a good idea and maintenance should be limited to funded work.

If your company requires a defined release cadence you are welcome to fund the project accordingly.

**Reply by adriangb** [reply]
That's totally fair, thank you for the response! I'm sorry if I came off as rude, it was not my intention, I did not expect anything, I only asked because I did not see any sort of public answer.

You are completely within your rights to guarantee nothing, especially for free, or have any other way of working that makes sense for you.

Thanks!
